lib/Frontend/ArgsToFrontendOptionsConverter.cpp

```diff
@@ -498,6 +503,40 @@ bool ArgsToFrontendOptionsConverter::setUpImmediateArgs() {
   return false;
 }
 
+bool ArgsToFrontendOptionsConverter::computeModuleAliases() {
+  auto list = Args.getAllArgValues(options::OPT_module_alias);
+  if (!list.empty()) {
+    for (auto val: list) {
+      auto i = val.find("=");
+      if (i == std::string::npos) {
+          Diags.diagnose(SourceLoc(), diag::error_bad_module_alias, val);
+          return true;
+      }
+
+      std::string *lhs = new std::string(val.substr(0, i));
+      std::string *rhs = new std::string(val.substr(i+1, val.length()));
```

Comment:
These `std::string`s will leak if the AST context is destroyed, won't they?

I think the best solution is to make `ModuleAliasMap` own its keys and values. That probably means it should be an `llvm::StringMap<std::string>`. (StringMap is a map with a custom implementation designed for variable-length string keys.)

With that in place, you can traffic in `StringRef`s up until it's time to insert the data into `ModuleAliasMap`—`list` will keep the underlying `std::string`s alive.